src/main/java/org/isf/patient/gui/PatientInsertExtended.java

```diff
@@ -417,11 +425,21 @@ private JButton getJAnamnesisButton() {
 				dialog.setModal(insert);
 				dialog.setVisible(true);
 			});
-		  
+
 		}
 		return jAnamnesisButton;
 	}
 
+	private void validateConsensus(PatientConsensus patientConsensus) throws OHDataValidationException {
+		List<OHExceptionMessage> errors = new ArrayList<>();
+		if (patientConsensus == null || !patient.getPatientConsensus().isConsensusFlag()) {
+			errors.add(new OHExceptionMessage(MessageBundle.getMessage("angal.patient.consensus.consensus.mandatory.msg")));
+		}
+		if (!errors.isEmpty()) {
+			throw new OHDataValidationException(errors);
+		}
+	}
```

Comment:
If the patient is a new patient this method generate a NPE. It should be moved to `patientConsensusManager` and so will be called after `patientBrowserManager.savePatient(patient);`